velox/exec/tests/NestedLoopJoinTest.cpp

```diff
@@ -622,5 +622,36 @@ TEST_F(NestedLoopJoinTest, mergeBuildVectors) {
   ASSERT_TRUE(waitForTaskCompletion(cursor->task().get()));
 }
 
+TEST_F(NestedLoopJoinTest, leftSemiJoinProjectDataValidation) {
+  auto probeVectors = makeRowVector(
+      {"t0"},
+      {sequence<int32_t>(5)});
+
+  auto buildVectors = makeRowVector(
+      {"u0"},
+      {sequence<int32_t>(3, 2)});
```

Comment:
Would you like to add some tests for data with null, and join with filter to make sure the behaviour is as expected?